base/linalg/dense.jl

```diff
@@ -29,6 +33,41 @@ function scale!(X::Array{T}, s::Real) where T<:BlasComplex
     X
 end
 
+
+function isone(x::StridedMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    if m < div(ISONE_CUTOFF * m * m, sizeof(x))
```

Comment:
`sizeof(x)` scales as `m^2` because it's a square matrix, so the size of an element is actually `sizeof(x) / m^2`.  I feel convinced that the two checks are equal, but you are definitely right that this should be simplified to `sizeof(x) < ISONE_CUTOFF * m` ... and when you look at it like that, it seems kind of obvious that `ISONE_CUTOFF` should just absorb the dimensionless `m`.  Would be really nice if this check could just be: `sizeof(x) < __CPU_CACHE_SIZE__`, or something like that.